src/backend/distributed/commands/multi_copy.c

```diff
@@ -2480,8 +2477,9 @@ ShardIdForTuple(CitusCopyDestReceiver *copyDest, Datum *columnValues, bool *colu
 	 * For reference table, this function blindly returns the tables single
 	 * shard.
 	 */
-	ShardInterval *shardInterval = FindShardInterval(partitionColumnValue,
-													 copyDest->tableMetadata);
```

Comment:
This definitely needs to be fixed. Any thoughts on whether it might affect performance?

Reply:
`hyperscale_tutorial` went from ~1550ms to ~1575ms on my machine

Reply:
Tested copying impressions.csv from hyperscale_tutorial test many times so that there was about 45 seconds of copying & truncating. Any impact was swallowed by noise. Usually the lookup will be a hash read & invalidation message check. It seems odd to try optimize caching a cache lookup, made sense in the initial implementation that had tableMetadata since there the receive calls were in a tight loop